src/main/java/Domain/Role.java

```diff
@@ -0,0 +1,237 @@
+package Domain;
+import java.util.HashSet;
+import java.util.Set;
+
+import Exceptions.*;
+
+/** 
+ * This class represents the role of a user in a specific shop.
+ */
+//TODO: ADD LOGS
+//TODO: ADD TESTS
```

Comment:
unit tests for domain, make sure to also add acceptance tests for service layer, and update the use-case doc if needed, and of course the drawio! :)

Reply:
thank you, fill free to write issues for that